File: Sources/XMTP/Messages/AuthData.swift
//
//  AuthData.swift
//
//
//  Created by Pat Nakajima on 11/17/22.
//

import Foundation

typealias AuthData = Xmtp_MessageApi_V1_AuthData

extension AuthData {
	init(walletAddress: String, timestamp: Date? = nil) {
		self.init()
		walletAddr = walletAddress

		let timestamp = timestamp ?? Date()
		createdNs = UInt64(timestamp.millisecondsSinceEpoch * 1_000_000)
	}
}

File: library/src/main/java/org/xmtp/android/library/messages/AuthData.kt
package org.xmtp.android.library.messages

import org.xmtp.proto.message.api.v1.Authn
import java.util.Date

typealias AuthData = org.xmtp.proto.message.api.v1.Authn.AuthData

class AuthDataBuilder {
    companion object {
        fun buildFromWalletAddress(walletAddress: String, timestamp: Date? = null): Authn.AuthData {
            val timestamped = timestamp?.time ?: System.currentTimeMillis()
            return AuthData.newBuilder().apply {
                walletAddr = walletAddress
                createdNs = timestamped * 1_000_000
            }.build()
        }
    }
}
